executor/insert_common.go

```diff
@@ -262,6 +271,20 @@ func insertRows(ctx context.Context, base insertCommon) (err error) {
 	return nil
 }
 
+func checkRowDoesNotMatchGivenPartitionSet(e *InsertValues, partitionTable table.PartitionedTable, row []types.Datum) error {
+	if partitionTable == nil {
+		return nil
+	}
+	p, err := partitionTable.GetPartitionByRow(e.ctx, row)
```

Comment:
This check is a kind of too heavy, you may see the implementation of `GetPartitionByRow`.
We already have some place calling `GetPartitionByRow`, could it be moved there?